src/lib/components/projects/ProjectActions.svelte

```diff
@@ -0,0 +1,98 @@
+<script context="module" lang="ts">
+  type Action = "edit" | "share" | "users" | "delete";
+</script>
+
+<script lang="ts">
+  import type { Project, ProjectUser } from "$lib/api/types";
+
+  import { _ } from "svelte-i18n";
+  import {
+    Alert16,
+    People16,
+    Pencil16,
+    Search16,
+    Share16,
+  } from "svelte-octicons";
+
+  import Flex from "../common/Flex.svelte";
+  import SidebarItem from "../sidebar/SidebarItem.svelte";
+
+  import Modal from "../layouts/Modal.svelte";
+  import Portal from "../layouts/Portal.svelte";
+
+  import EditProject from "../forms/EditProject.svelte";
+  import ManageCollaborators from "../forms/ManageCollaborators.svelte";
+  import DeleteProject from "../forms/DeleteProject.svelte";
+
+  import { projectSearchUrl } from "$lib/utils/search";
+
+  export let project: Project;
+  export let users: ProjectUser[];
+
+  let show: Action = null;
+
+  const actions: Record<Action, string> = {
+    edit: $_("projects.edit"),
+    share: $_("projects.share"),
+    users: $_("projects.users"),
+    delete: $_("projects.delete.confirm"),
+  };
+
+  function hide() {
+    show = null;
+  }
+</script>
+
+<Flex direction="column">
+  <SidebarItem hover on:click={() => (show = "edit")}>
+    <Pencil16 />{$_("sidebar.edit")}
+  </SidebarItem>
+
+  <SidebarItem hover on:click={() => (show = "users")}>
+    <People16 />{$_("sidebar.collaborate")}
+  </SidebarItem>
+
+  <SidebarItem hover on:click={() => (show = "share")}>
+    <Share16 />{$_("sidebar.shareEmbed")}
+  </SidebarItem>
+  <SidebarItem
+    hover
+    --color="var(--caution)"
+    --fill="var(--caution)"
+    on:click={() => (show = "delete")}
+  >
+    <Alert16 />
+    {$_("projects.delete.action")}
+  </SidebarItem>
+</Flex>
+
+<hr class="divider" />
+
+<SidebarItem href={projectSearchUrl(project)}>
+  <Search16 />{$_("projects.viewInSearch")}
+</SidebarItem>
+
+{#if show}
+  <Portal>
+    <Modal on:close={hide}>
+      <h1 slot="title">
+        {actions[show]}
+      </h1>
+      {#if show === "edit"}
+        <EditProject {project} on:close={hide} />
+      {/if}
+
+      {#if show === "share"}
+        <p>Share</p>
+      {/if}
+
+      {#if show === "users"}
+        <ManageCollaborators {project} {users} on:close={hide} />
+      {/if}
+
+      {#if show === "delete"}
+        <DeleteProject {project} on:close={hide} />
+      {/if}
+    </Modal>
+  </Portal>
+{/if}
```

Comment:
For the record, I just want to flag why this approach is valid and an improvement over what came before, even though it uses conditional logic to decide what to show in the modal. Previously, all modals had to be declared in a top level layout component based on global conditional flags. Here, the modal is rendered in-context of the UI that is triggering it, based on a limited set of user interactions. If we need to add or remove an interaction, all of that can happen within a single file.